pkg/cloud/aws/services/ec2/ami.go

```diff
@@ -16,43 +16,53 @@ limitations under the License.
 
 package ec2
 
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/klog"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/pkg/errors"
+)
+
+func amiName(platform, platformVersion, kubernetesVersion string) string {
+	return fmt.Sprintf("ami-%s-%s-%s-00-??????????", platform, platformVersion, strings.TrimPrefix(kubernetesVersion, "v"))
+}
+
 // defaultAMILookup returns the default AMI based on region
-func (s *Service) defaultAMILookup(region string) string {
-	//TODO(chuckha) Replace this function with a method to filter public images.
-	switch region {
-	case "ap-northeast-1":
-		return "ami-06525773f0332c0ca"
-	case "ap-northeast-2":
-		return "ami-0914a816a80ab8779"
-	case "ap-south-1":
-		return "ami-04b17f6875b4d9c29"
-	case "ap-southeast-1":
-		return "ami-0ccaa193408259c82"
-	case "ap-southeast-2":
-		return "ami-0385d030f1b7df12c"
-	case "ca-central-1":
-		return "ami-031ca32942ffbe25f"
-	case "eu-central-1":
-		return "ami-078f9fe8bdaa81aa8"
-	case "eu-west-1":
-		return "ami-09547cd6f9856a79b"
-	case "eu-west-2":
-		return "ami-08efe1a8b5f18f381"
-	case "eu-west-3":
-		return "ami-042c578ede4ff3f33"
-	case "sa-east-1":
-		return "ami-0dd7e2894db509204"
-	case "us-east-1":
-		return "ami-026e9f3a713727945"
-	case "us-east-2":
-		return "ami-0ec6d3241fb7776fe"
-	case "us-west-1":
-		return "ami-06ec1c533176de131"
-	case "us-west-2":
-		return "ami-0cfa2d1fa5cc93615"
-	default:
-		return "unknown region"
+func (s *Service) defaultAMILookup(platform, platformVersion, kubernetesVersion string) (string, error) {
+	describeImageInput := &ec2.DescribeImagesInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("name"),
+				Values: []*string{aws.String(amiName(platform, platformVersion, kubernetesVersion))},
+			},
+			{
+				Name:   aws.String("architecture"),
+				Values: []*string{aws.String("x86_64")},
+			},
+			{
+				Name:   aws.String("state"),
+				Values: []*string{aws.String("available")},
+			},
+			{
+				Name:   aws.String("virtualization-type"),
+				Values: []*string{aws.String("hvm")},
+			},
```

Comment:
We need to hardcode the account id as a filter, based on these settings someone could hijack the same naming and filters and publish their own default AMI. AWS returns the most recent one.  @chuckha